crates/bevy_reflect/src/reflect.rs

```diff
@@ -279,8 +307,10 @@ impl dyn Reflect {
     ///
     /// Read `is` for more information on underlying values and represented types.
     #[inline]
-    pub fn represents<T: Reflect>(&self) -> bool {
-        self.type_name() == any::type_name::<T>()
+    pub fn represents<T: Reflect + TypePath>(&self) -> bool {
+        self.get_represented_type_info()
+            .map(|t| t.type_path() == T::type_path())
+            .unwrap_or(false)
```

Comment:
Should this be using `DynamicTypePath`? Or does that not work for `dyn Reflect`?
```suggestion
        self.reflect_type_path() == T::type_path()
```

Reply:
that would give it the same behaviour as `is`. it would not respect the represented types of `DynamicFoo`s.

Reply:
Oh right. Good catch!